### ISSUE TITLE
404 page not found when credentials from metadata service in --server mode

### ISSUE THREAD

**Issue by MasterNayru** [99designs/aws-vault#577]
- [x] I am using the latest release of AWS Vault
- [x] I have provided my `.aws/config` (redacted if necessary)
- [x] I have provided the debug output using `aws-vault --debug` (redacted if necessary)

It would appear that with v5.4.2, I am unable to credentials from the metadata service spawned in --server mode.

```$ aws s3 ls s3://        
2020/05/04 16:01:34 http: 127.0.0.1:54684: 404 GET / (29.11µs)
Unable to locate credentials. You can configure credentials by running "aws configure".
```

The service seems to start up fine, but I get a 404 when I try to curl the metadata service for credentials:
```$ curl http://169.254.169.254/latest/meta-data/iam/security-credentials/local-credentials
2020/05/04 15:53:28 http: 127.0.0.1:52564: 404 GET / (24.22µs)
404 page not found
```
However, if I hit 127.0.0.1:9099, it does seem like I can get credentials back:
```
$ curl http://127.0.0.1:9099/latest/meta-data/iam/security-credentials/local-credentials
2020/05/04 15:54:26 Credentials.IsExpired() = false
2020/05/04 15:54:26 Serving credentials via http ****************GVQE, expiration of 2020-05-04T13:47:38Z (7h53m11.770975844s)
2020/05/04 15:54:26 http: 127.0.0.1:52822: 200 GET /latest/meta-data/iam/security-credentials/local-credentials (140.789µs)
{"AccessKeyId":"REDACTED","Code":"Success","Expiration":"2020-05-04T13:47:38Z","LastUpdated":"2020-05-04T15:54:26Z","SecretAccessKey":"REDACTED","Token":"REDACTED","Type":"AWS-HMAC"}
```
I am able to reproduce this behaviour in the beta versions of v6 and v5.4.2, but v5.3.2 does not seem to have this issue.

Redacted ~/.aws/config
```[profile bastion]
region = us-east-2

[profile development]
region = us-east-2
source_profile = bastion
mfa_serial = arn:aws:iam::12345678910:mfa/masternayru
role_arn = arn:aws:iam::12345678910:role/myrole
duration_seconds = 28800
```

Debug output:
```
$ aws-vault --backend file exec development --server --duration=8h --debug
2020/05/04 15:52:22 aws-vault v5.4.2
2020/05/04 15:52:22 [keyring] Considering backends: [file]
2020/05/04 15:52:22 Loading config file /home/jmo/.aws/config
2020/05/04 15:52:22 Parsing config file /home/jmo/.aws/config
2020/05/04 15:52:22 Using region "us-east-2" from AWS_DEFAULT_REGION
2020/05/04 15:52:22 Using region "us-east-2" from AWS_DEFAULT_REGION
2020/05/04 15:52:22 [keyring] Expanded file dir to /home/jmo/.awsvault/keys/
2020/05/04 15:52:22 [keyring] Expanded file dir to /home/jmo/.awsvault/keys/
2020/05/04 15:52:22 profile bastion: using stored credentials
2020/05/04 15:52:22 profile bastion: skipping GetSessionToken because MFA serial doesn't match profile 'development'
2020/05/04 15:52:22 profile development: using AssumeRole (with MFA)
Enter token for arn:aws:iam::12345678910:mfa/masternayru: 123456
2020/05/04 15:52:34 Looking up keyring for 'bastion'
2020/05/04 15:52:34 [keyring] Expanded file dir to /home/jmo/.awsvault/keys/
2020/05/04 15:52:34 [keyring] Expanded file dir to /home/jmo/.awsvault/keys/
Enter passphrase to unlock /home/jmo/.awsvault/keys/: 
2020/05/04 15:52:38 Generated credentials ****************GVQE using AssumeRole, expires in 7h59m59.478942006s
2020/05/04 15:52:38 Setting subprocess env: AWS_DEFAULT_REGION=us-east-2, AWS_REGION=us-east-2
2020/05/04 15:52:38 Starting child process: /bin/zsh 
2020/05/04 15:52:38 Starting EC2 Instance Metadata server on 127.0.0.1:9099
```

**Comment by mtibben** [comment]
@MasterNayru can you provide debug logs using v6.0.0-beta4?

**Comment by mtibben** [comment]
Also please kill any old background proxy process using `sudo killall -9 aws-vault` before running the command

**Comment by MasterNayru** [comment]
@mtibben Here is the debug log I get for the server process itself:
```$ aws-vault --backend file exec development --server --duration=8h --debug                                                           
2020/05/04 16:22:01 aws-vault v6.0.0-beta4                                                                                              
2020/05/04 16:22:01 Loading config file /home/jmo/.aws/config                                                                           
2020/05/04 16:22:01 Parsing config file /home/jmo/.aws/config                                                                           
2020/05/04 16:22:01 [keyring] Considering backends: [file]                                                                              
2020/05/04 16:22:01 Using region "us-east-2" from AWS_DEFAULT_REGION                                                                    
2020/05/04 16:22:01 Profile 'default' missing in config file                                                                            
2020/05/04 16:22:01 Using region "us-east-2" from AWS_DEFAULT_REGION                                                                    
2020/05/04 16:22:01 Profile 'default' missing in config file                                                                            
2020/05/04 16:22:01 [keyring] Expanded file dir to /home/jmo/.awsvault/keys/
2020/05/04 16:22:01 [keyring] Expanded file dir to /home/jmo/.awsvault/keys/
2020/05/04 16:22:01 profile bastion: using stored credentials
2020/05/04 16:22:01 profile bastion: skipping GetSessionToken because MFA serial doesn't match profile 'development'
2020/05/04 16:22:01 profile development: using AssumeRole (with MFA) 
2020/05/04 16:22:01 Starting `aws-vault server` as root in the background
[sudo] password for jmo: 
2020/05/04 16:22:06 [keyring] Expanded file dir to /home/jmo/.awsvault/keys/
2020/05/04 16:22:06 [keyring] Expanded file dir to /home/jmo/.awsvault/keys/
2020/05/04 16:22:06 [keyring] Expanded file dir to /home/jmo/.awsvault/keys/
2020/05/04 16:22:06 [keyring] Expanded file dir to /home/jmo/.awsvault/keys/
Enter passphrase to unlock /home/jmo/.awsvault/keys/: aws-vault: error: server: exit status 2

2020/05/04 16:22:11 Re-using cached credentials ****************IPXQ from sts.AssumeRole, expires in 7h11m46.958378814s
2020/05/04 16:22:11 Setting subprocess env: AWS_DEFAULT_REGION=us-east-2, AWS_REGION=us-east-2
2020/05/04 16:22:11 Starting child process: /bin/zsh 
2020/05/04 16:22:11 Starting EC2 Instance Metadata server on 127.0.0.1:9099
```
The `aws-vault: error: server: exit status 2` bit in the output is a bit weird, but typing in my aws-vault passphrase and hitting Enter seemed to allow the process to start up.

Then for the curl commands, I get this:
```
$ curl http://169.254.169.254/latest/meta-data/iam/security-credentials/local-credentials
curl: (7) Failed to connect to 169.254.169.254 port 80: Connection refused
$ curl http://127.0.0.1:9099/latest/meta-data/iam/security-credentials/local-credentials 
2020/05/04 16:22:19 Credentials.IsExpired() = false
2020/05/04 16:22:19 Serving credentials via http ****************IPXQ, expiration of 2020-05-04T13:28:58Z (7h6m38.084211823s)
2020/05/04 16:22:19 http: 127.0.0.1:60256: 200 GET /latest/meta-data/iam/security-credentials/local-credentials (121.23µs)
{"AccessKeyId":"REDACTED","Code":"Success","Expiration":"2020-05-04T13:28:58Z","LastUpdated":"2020-05-04T16:22:19Z","SecretAccessKey":"REDACTED","Token":"REDACTED","Type":"AWS-HMAC"}
```

**Comment by mtibben** [comment]
`aws-vault: error: server: exit status 2` indicates the background proxy process may be failing to start. What OS are you on?

**Comment by MasterNayru** [comment]
Fedora 32. I have a co-worker experiencing the issue on macOS with the same set of versions, in case that helps.

**Comment by mtibben** [comment]
The changes to the ec2 metadata server was made in v5.4.0 in support of the hidden and experimental `--ecs-server` flag. These changes were supposed to have been benign refactoring, but seems to be causing some issues. So I've reverted the changes and released [v5.4.3](https://github.com/99designs/aws-vault/releases/tag/v5.4.3).

The problems may still exist on master and the v6 beta, although I can't reproduce this issue directly yet

**Comment by MasterNayru** [comment]
@mtibben I have tested v5.4.3 locally on my Fedora 32 machine and can confirm that it is working now. If you need any more debug info on my side to help diagnose this issue, please let me know as it sounds like that ECS metadata service feature will be the preferred approach going forward.

Will close this issue as my immediate issue seems to have been resolved.
